packages/host/app/services/store.ts

```diff
@@ -121,16 +121,26 @@ export default class StoreService extends Service implements StoreInterface {
       return;
     }
     let currentReferenceCount = this.referenceCount.get(id) ?? 0;
-    currentReferenceCount--;
+    currentReferenceCount -= 1;
     this.referenceCount.set(id, currentReferenceCount);
 
-    if (currentReferenceCount === 0) {
+    console.debug(
+      `dropping reference to ${id}, current reference count: ${this.referenceCount.get(id)}`,
+    );
+    if (currentReferenceCount <= 0) {
       this.referenceCount.delete(id);
       let autoSaveState = this.autoSaveStates.get(id);
       if (autoSaveState?.hasUnsavedChanges) {
         this.initiateAutoSaveTask.perform(id, { isImmediate: true });
       }
-      this.autoSaveStates.delete(id);
+      // await for a microtask to prevent rerender dirty tag error so we don't
+      // get in trouble because we read this.autosaveStates in the same frame as
+      // we mutate this.autosaveStates
+      (async () => {
```

Comment:
Is this solving the type of that ember error that goes something like `You attempted to update ... on
... but it had already been used previously in the same computation.`? Just curious

Reply:
yup. you cant read and update a tracked property in the same frame, as you may be different results based on whether you read first or set first.. so this makes it so they are separated by the smallest increment of time possible so that reading and setting are deterministic

Reply:
that is a totally reasonable approach as well